include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1067,17 +1070,17 @@ struct __early_exit_find_or
             if (__shifted_idx < __n && __pred(__shifted_idx, __rngs...))
             {
                 if constexpr (_OrTagType::value)
-                {
                     __found_local.store(1);
-                    break;
-                }
                 else
                 {
                     for (auto __old = __found_local.load(); __comp(__shifted_idx, __old); __old = __found_local.load())
```

Comment:
I wonder if this combination of two atomic operations (load then exchange) can actually be converted to a single `fetch_min` or `fetch_max` followed by non-atomic local check of the return vs the operand to check if early exit can be triggered.  This would also remove the possibility to "miss" which is what requires the loop in my understanding.

I believe `__comp` will always be `less` or `greater` in this branch, and no more exotic comparisons are required, which should enable such a change.

If this isn't quite right, then I'd consider the weak exchange comment here instead..

Reply:
About the usage `fetch_min` and `fetch_max` - looks like it's give about additional 10% so I will apply it.
Good catch, @danhoeflinger, thanks!

Reply:
Fixed.